### PR TITLE
Fix minitest framework and add additional helpers

### PR DESCRIPTION
**Pull request by alexcu**
@jakerenzella and myself worked on fixing the test framework to ensure all tests use minitest and that transactions are used on our [new/helpdesk-ticketing](https://github.com/final-year-project/doubtfire-api/tree/new/helpdesk-ticketing) branch. This pull request cherry-picks those commits into this branch so that it can be implemented in [develop](https://github.com/doubtfire-api/doubtfire-api/tree/develop)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -48,7 +48,8 @@ group :development, :test, :replica do
   gem 'minitest-rails'
   gem 'minitest-hyper'
   gem 'database_cleaner'
-  gem "minitest-osx"
+  gem 'minitest-osx'
+  gem 'minitest-around'
```
Comment (alexcu) [inline Gemfile]: @jakerenzella Did we actually end up using `minitest-around`? I remember we were trying it out for something but that didn't work... do we still need it?

**Comment by jakerenzella** [comment]
@macite Please do not merge... further commits on their way :)

**Comment by jakerenzella** [comment]
Ready to merge :)

**Comment by macite** [comment]
Looks good.

**Pull request merged**